ext/socket/rubysocket.h

```diff
@@ -344,6 +344,10 @@ socklen_t rsock_unix_sockaddr_len(VALUE path);
 
 int rsock_socket(int domain, int type, int proto);
 int rsock_detect_cloexec(int fd);
+
+void rsock_validate_descriptor(int fd);
```

Comment:
Is this needed?

Reply:
No, we can remove it. I also thought we could move everything to just the one place it's used, in `bsock_s_for_fd`.

Reply:
Okay, updated PR.